lib/std/heap.zig

```diff
@@ -8,6 +8,273 @@ const c = std.c;
+        .csky => 4 << 10,
+        // Hexagon's Kconfig/page.h in Linux. Non-huge pages go past 256KB.
+        .hexagon => 4 << 10,
+        // Source: Zig's own libc page.h for arc.
+        .arc => 4 << 10,
+        // Source: Wikipedia "Page (computer memory)"
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 4 << 10,
+        .riscv32, .riscv64 => 4 << 10,
+        .sparc => 4 << 10,
+        .sparc64 => 8 << 10,
+        else => missing_min_page_size,
+    },
+    else => missing_min_page_size,
+};
+
+/// This value defines the largest page size for this architecture/OS combination that the standard library allows. The standard library asserts that `pageSize()` does not exceed `max_page_size`. Using a larger page size requires modifying an appropriate prong in the definition of `max_page_size`. See also the linker argument `-z max-page-size=`.
+pub const max_page_size: usize = switch (builtin.os.tag) {
+    .driverkit, .ios, .macos, .tvos, .visionos, .watchos => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        else => missing_max_page_size,
+    },
+    // -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+    .windows => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        else => missing_max_page_size,
+    },
+    .wasi => switch (builtin.cpu.arch) {
+        .wasm32, .wasm64 => 64 << 10,
+        else => missing_max_page_size,
+    },
+    // https://github.com/tianocore/edk2/blob/b158dad150bf02879668f72ce306445250838201/MdePkg/Include/Uefi/UefiBaseType.h#L180-L187
+    .uefi => 4 << 10,
+    .freestanding => switch (builtin.cpu.arch) {
+        .wasm32, .wasm64 => 64 << 10,
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 64 << 10,
+        // Explicitly only 4kb.
+        // https://refspecs.linuxbase.org/ELF/zSeries/lzsabi0_zSeries.html#AEN798
+        .s390x => 4 << 10,
+        // Source: Zig's own libc page.h for arc.
+        .arc => 16 << 10,
+        // Source: Wikipedia "Page (computer memory)"
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 64 << 10,
+        .riscv32, .riscv64 => 4 << 10,
+        .sparc => 256 << 10,
+        .sparc64 => 64 << 10,
+        else => missing_max_page_size,
+    },
+    .freebsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // https://github.com/freebsd/freebsd-src/blob/0a9d1da6e6cede5e9c0ff63240d724049ad72b5b/sys/arm64/arm64/locore.S#L931
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
```

Comment:
arm/thumb use 4K.

aarch64 is... confusing. I see code that handles 4K, 16K, and 64K. 64K support seems incomplete, per your link. But `param.h` hardcodes it to 4K anyway. I suggest just using 4K until we have evidence that 16K or 64K are actually formally supported.